src/main/java/ai/Storage.java

```diff
@@ -0,0 +1,103 @@
+package ai;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import ai.exception.AiException;
+import ai.task.Deadline;
+import ai.task.Event;
+import ai.task.Task;
+import ai.task.ToDo;
+
+/**
+ * Deals with file writing and reading.
+ */
+public class Storage {
+    private ArrayList<Task> tasks = new ArrayList<Task>();
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the file using the given filePath and scans for each line.
+     *
+     * @return ArrayList of type Task.
+     * @throws IOException if .txt file cannot be created successfully.
+     * @throws AiException if the line in the .txt file cannot be read.
+     */
+    public ArrayList<Task> load() throws IOException, AiException {
+        Path path = Paths.get(filePath);
+        Path directory = path.getParent();
```

Comment:
Maybe the name can be more descriptive, not to be mistaken with the actual directory file

```suggestion
        Path directoryPath = path.getParent();
```